external-plugins/rehearse/plugin/handler/handler.go

```diff
@@ -0,0 +1,504 @@
+	basePresubmits := hashPresubmitsConfig(baseConfig.PresubmitsStatic)
+
+	for presubmitKey, headPresubmit := range headPresubmits {
+		basePresubmit, exists := basePresubmits[presubmitKey]
+
+		if exists && reflect.DeepEqual(basePresubmit.Spec, headPresubmit.Spec) {
+			continue
+		}
+		// This is a new or modified presubmit
+		// TODO: check the repo configuration on the generated Prow job - will it report the correct place?
+		log.Infof("Detected modified presubmit: %s.", headPresubmit.Name)
+
+		job := pjutil.NewPresubmit(*pr, pr.Base.SHA, headPresubmit, eventGUID)
+
+		repoOrg := repoFromJobKey(presubmitKey)
+		org, repo, err := gitv2.OrgRepo(repoOrg)
+		if err != nil {
+			log.Errorf(
+				"Could not extract repo and org from job key: %s. Job name: %s",
+				presubmitKey, headPresubmit.Name)
+		}
+
+		job.Spec.ExtraRefs = addExtraRefs(job.Spec.ExtraRefs, org, repo, pr.Base.Ref)
+		jobs = append(jobs, job)
+	}
+	return jobs
+}
+
+func (h *gitHubEventsHandler) generatePeriodics(headConfig, baseConfig *config.Config) []prowapi.ProwJob {
+	var jobs []prowapi.ProwJob
+
+	headPeriodics := hashPeriodicsConfig(headConfig.AllPeriodics())
+	basePeriodics := hashPeriodicsConfig(baseConfig.AllPeriodics())
+
+	for periodicKey, headPeriodic := range headPeriodics {
+		basePeriodic, exists := basePeriodics[periodicKey]
+		if !exists || !reflect.DeepEqual(basePeriodic, headPeriodic) {
+			// This is a new or modified periodic
+			// TODO: check the repo configuration on the generated Prow job - will it report the correct place?
+			log.Infof("Detected modified periodic: %s.", headPeriodic.Name)
+			spec := pjutil.PeriodicSpec(headPeriodic)
+
+			repoFromJobKey(periodicKey)
+			job := pjutil.NewProwJob(spec, nil, nil)
+			job.Spec.Report = true
+			jobs = append(jobs, job)
+		}
+	}
+
+	return jobs
+}
+
+func (h *gitHubEventsHandler) loadConfigsAtRef(
+	changedJobConfigs []string, git gitv2.RepoClient, ref string) (map[string]*config.Config, error) {
+	configs := map[string]*config.Config{}
+
+	tmpdir, err := ioutil.TempDir("", "prow-configs")
+	if err != nil {
+		log.WithError(err).Error("Could not create a temp directory to store configs.")
+		return nil, err
+	}
+	defer os.RemoveAll(tmpdir)
+	// In order to actually support multi-threaded access to the git repo, we can't checkout any refs or assume that
+	// the repo is checked out with any refspec. Instead, we use git cat-file to read the files that we need to the
+	// memory and write them to a temp file. We need the temp file because the current version of Prow's config module
+	// can't load the configs from the memory.
+	prowConfigBytes, ret := catFile(log, git.Directory(), h.prowConfigPath, ref)
+	if ret != 0 && ret != 128 {
+		log.WithError(err).Errorf("Could not load Prow config [%s] at ref: %s", h.prowConfigPath, ref)
+		return nil, err
+	}
+	log.Infoln("File:", string(prowConfigBytes))
+	prowConfigTmp, err := writeTempFile(log, tmpdir, prowConfigBytes)
```

Comment:
Will this be clean up too?

Reply:
Yes, when the directory will be removed, the files in it will be removed as well.
See https://github.com/kubevirt/project-infra/pull/554/files#diff-1cf555f4744be30bc95023fc222bd9d3R354